docs/dev/libs/state_processor_api.md

```diff
@@ -0,0 +1,459 @@
+        Collector<Transaction> out) throws Exception {
+ 
+     Double rate = ctx.getBroadcastState(CURRENCY_RATES).get(value.currencyId);
+     if (rate != null) {
+        value.amount *= rate;
+     }
+     out.collect(value);
+  }
+ 
+  public void processBroadcastElement(
+        CurrencyRate value,
+        Context ctx,
+        Collector<Transaction> out) throws Exception {
+        ctx.getBroadcastState(CURRENCY_RATES).put(value.currencyId, value.rate);
+  }
+}
+  
+class Summarize extends RichFlatMapFunction<Transaction, Summary> {
+  transient ValueState<Double> totalState;
+  transient ValueState<Integer> countState;
+ 
+  public void open(Configuration configuration) throws Exception {
+     totalState = getRuntimeContext().getState(new ValueStateDescriptor<>("total", Types.DOUBLE));
+     countState = getRuntimeContext().getState(new ValueStateDescriptor<>("count", Types.INT));
+  }
+ 
+  public void flatMap(Transaction value, Collector<Summary> out) throws Exception {
+     Summary summary = new Summary();
+     summary.total = value.amount;
+     summary.count = 1;
+ 
+     Double currentTotal = totalState.value();
+     if (currentTotal != null) {
+        summary.total += currentTotal;
+     }
+ 
+     Integer currentCount = countState.value();
+     if (currentCount != null) {
+        summary.count += currentCount;
+     }
+     countState.update(summary.count);
+ 
+     out.collect(summary);
+  }
+}
+ 
+DataStream<Transaction> transactions = . . .
+BroadcastStream<CurrencyRate> rates = . . .
+transactions
+  .connect(rates)
+  .process(new CurrencyConverter())
+  .uid("currency_converter")
+  .keyBy(transaction -> transaction.accountId)
+  .flatMap(new Summarize())
+  .uid("summarize")
+{% endhighlight %}
+</div>
+
+This job contains multiple operators along with various kinds of state.
+When analyzing that state we can first scope data by its operator, named by setting its uid.
+Within each operator we can look at the registered states.
+`CurrencyConverter` has a broadcast state, which is a type of non-partitioned operator state.
+In general, there is no relationship between any two elements in an operator state and so we can look at each value as being its own row.
+Contrast this with Summarize, which contains two keyed states.
+Because both states are scoped under the same key we can safely assume there exists some relationship between the two values.
+Therefore, keyed state is best understood as a single table per operator containing one _key_ column along with _n_ value columns, one for each registered state.
+All of this means that the state for this job could be described using the following pseudo-sql commands. 
+
+{% highlight sql %}
+CREATE NAMESPACE currency_converter;
+ 
+CREATE TABLE currency_converter.rates (
+   value Tuple2<Integer, Double>
+);
+ 
+CREATE NAMESPACE summarize;
+ 
+CREATE TABLE summarize.keyed_state (
+   key   INTEGER PRIMARY KEY,
+   total DOUBLE,
+   count INTEGER
```

Comment:
`count` would be treated as a key word for SQL and showed as bold in web UI, shall we rename this to another name?

Reply:
I put it and key in backticks, it's more about expressing the idea.